sdk/core/Azure.Core.Experimental/src/Variant/Value.Union.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Azure {
+    public readonly partial struct Value
```

Comment:
A comment on the implementation details would be nice here.

Reply:
I will improve up the comments and docs in a subsequent PR once the implementation/APIs stabilize.